### PR TITLE
Syw UID2-3502 fix android doco

### PR DESCRIPTION
**Pull request by sunnywu**
1.  add more info (R8/Proguard warning) from android sdk repo readme into IMA/GMA plugin guide,
2. change minimum requirements for different uses of SDK and make it clearer to app publishers

### CODE REVIEW

File: docs/sdks/uid2-sdk-ref-android.md [hunk 1]
```diff
@@ -94,7 +99,7 @@ There are two options for installing the Android UID2 SDK:
 
 -   [Gradle](#installing-with-gradle)
 
--  [ Maven](#installing-with-maven)
+-  [Maven](#installing-with-maven)
```
Comment (sunnywu) [inline docs/sdks/uid2-sdk-ref-android.md:102]: previous typo

File: docs/sdks/uid2-sdk-ref-android.md [hunk 2]
```diff
@@ -81,9 +81,14 @@ The binary is published on Sonatype:
...
 ## Minimum Requirements
 
-Minimum requirements for this SDK are as follows:
+To consume the binary package of this SDK in your app:
 
-- Android Studio version: v2023.3.1+
+- Minimum target Android version: 4.4+ / API 19+ (SDK) 5.0+
+
+
+To run the development app as mentioned in the [Client-Side Integration Guide for Mobile](../guides/integration-mobile-client-side#client-side-integration-example) or build binary from source code, minimum requirements are as follows:
```
Comment (IanDBird) [inline docs/sdks/uid2-sdk-ref-android.md:89]: Good clarification, thanks!

**Review by IanDBird — APPROVED** [review]
Looks great, thanks!

File: docs/guides/mobile-plugin-ima-android.md [hunk 3]
```diff
@@ -67,3 +68,11 @@ To install with Maven, add the SDK as a dependency in the `pom.xml` file:
   <version>0.5.0</version>
 </dependency>
 ```
+
+## Notes for using R8 or ProGuard
+
+If you are using R8 the shrinking and obfuscation rules are included automatically.
```
Comment (genwhittTTD) [inline docs/guides/mobile-plugin-ima-android.md:74]: If you are using R8, the shrinking and obfuscation rules are included automatically.

File: docs/guides/mobile-plugin-ima-android.md [hunk 4]
```diff
@@ -67,3 +68,11 @@ To install with Maven, add the SDK as a dependency in the `pom.xml` file:
...
+
+ProGuard users must manually add the options from
+[uid2-ima.pro](https://github.com/IABTechLab/uid2-android-sdk/blob/main/securesignals-ima/uid2-ima.pro).
```
Comment (genwhittTTD) [inline docs/guides/mobile-plugin-ima-android.md:77]: If you are using ProGuard, you must manually add the option specified in the following file: [uid2-ima.pro](https://github.com/IABTechLab/uid2-android-sdk/blob/main/securesignals-ima/uid2-ima.pro).

File: docs/sdks/uid2-sdk-ref-android.md (hunk 2 above)
Comment (genwhittTTD) [inline docs/sdks/uid2-sdk-ref-android.md:89]: To run the development app (see [Client-Side Integration Guide for Mobile](../guides/integration-mobile-client-side#client-side-integration-example)), or to build the binary from source code, the minimum requirements are as follows:

File: docs/sdks/uid2-sdk-ref-android.md [hunk 5]
```diff
@@ -81,9 +81,14 @@ The binary is published on Sonatype:
...
+
+- Android Studio version: Check the Android Gradle Plugin (AGP) version required by the UID2 SDK specified in the [code repository](https://github.com/IABTechLab/uid2-android-sdk/blob/main/gradle/libs.versions.toml), and check [here](https://developer.android.com/build/releases/gradle-plugin) for the corresponding Android Studio version required
```
Comment (genwhittTTD) [inline docs/sdks/uid2-sdk-ref-android.md:91]: - Android Studio version: Check the Android Gradle Plugin (AGP) version required by the UID2 SDK specified in the [code repository](https://github.com/IABTechLab/uid2-android-sdk/blob/main/gradle/libs.versions.toml), and check [the Android Gradle release notes](https://developer.android.com/build/releases/gradle-plugin) for the corresponding Android Studio version required.

**Review by genwhittTTD — APPROVED** [review]
Few copy mods. Your call. LMK if any questions.

File: docs/guides/integration-mobile-client-side.md [hunk 6]
```diff
@@ -77,7 +77,7 @@ Follow the applicable instructions, for Android or iOS:
 <TabItem value='android' label='Android'>
 
 1. Check out the main branch of the [UID2 SDK for Android source code repository on GitHub](https://github.com/IABTechLab/uid2-android-sdk/tree/main).
-1. In Android Studio (Jellyfish/v2023.3.1 or whichever future version supports the Android Gradle Plugin version required by the UID2 SDK for Android release at the time), open the directory that you checked out.
+1. In Android Studio (check the version required in the [Minimum Requirements section](../sdks/uid2-sdk-ref-android.md#minimum-requirements) in UID2 SDK for Android Reference Guide, open the directory that you checked out.
```
[the remark this answers is not in the document]
Comment (sunnywu) [reply docs/guides/integration-mobile-client-side.md:80]: fixed

File: docs/guides/mobile-plugin-gma-android.md [hunk 7]
```diff
@@ -38,6 +38,7 @@ To run this plugin, install the following:
    - [SDK](https://central.sonatype.com/artifact/com.uid2/uid2-android-sdk)
    - [UID2 SDK for Android Reference Guide](../sdks/uid2-sdk-ref-android.md)
 1. [UID2 Android GMA Plugin v0.5.0](https://central.sonatype.com/artifact/com.uid2/uid2-android-sdk-gma/)
+1. If you are using R8 or Proguard, add the necessary file mentioned in [here](#notes-for-using-r8-or-proguard)
```
[the remark this answers is not in the document]
Comment (sunnywu) [reply docs/guides/mobile-plugin-gma-android.md:41]: fixed

File: docs/guides/mobile-plugin-gma-android.md [hunk 8]
```diff
@@ -67,3 +68,11 @@ To install with Maven, add the SDK as a dependency in the `pom.xml` file:
   <version>0.5.0</version>
 </dependency>
 ```
+
+## Notes for using R8 or ProGuard
```
[the remark this answers is not in the document]
Comment (sunnywu) [reply docs/guides/mobile-plugin-gma-android.md:72]: fixed

File: docs/guides/mobile-plugin-gma-android.md [hunk 9]
```diff
@@ -67,3 +68,11 @@ To install with Maven, add the SDK as a dependency in the `pom.xml` file:
...
+
+If you are using R8 the shrinking and obfuscation rules are included automatically.
```
[the remark this answers is not in the document]
Comment (sunnywu) [reply docs/guides/mobile-plugin-gma-android.md:74]: fixed

File: docs/guides/mobile-plugin-ima-android.md [hunk 10]
```diff
@@ -38,6 +38,7 @@ To run this plugin, install the following:
    - [SDK](https://central.sonatype.com/artifact/com.uid2/uid2-android-sdk)
    - [UID2 SDK for Android Reference Guide](../sdks/uid2-sdk-ref-android.md)
 1. [UID2 IMA Plugin for Android v0.5.0](https://central.sonatype.com/artifact/com.uid2/uid2-android-sdk-ima)
+1. If you are using R8 or Proguard, add the necessary file mentioned in [here](#notes-for-using-r8-or-proguard)
```
[the remark this answers is not in the document]
Comment (sunnywu) [reply docs/guides/mobile-plugin-ima-android.md:41]: fixed

File: docs/guides/mobile-plugin-ima-android.md (hunk 3 above)
Comment (sunnywu) [reply docs/guides/mobile-plugin-ima-android.md:74]: fixed

File: docs/guides/mobile-plugin-ima-android.md (hunk 4 above)
Comment (sunnywu) [reply docs/guides/mobile-plugin-ima-android.md:77]: fixed

File: docs/sdks/uid2-sdk-ref-android.md (hunk 2 above)
Comment (sunnywu) [reply docs/sdks/uid2-sdk-ref-android.md:89]: fixed

File: docs/sdks/uid2-sdk-ref-android.md (hunk 5 above)
Comment (sunnywu) [reply docs/sdks/uid2-sdk-ref-android.md:91]: fixed

File: docs/guides/mobile-plugin-gma-android.md [hunk 11]
```diff
@@ -67,3 +68,11 @@ To install with Maven, add the SDK as a dependency in the `pom.xml` file:
...
+
+ProGuard users must manually add the options from
```
[the remark this answers is not in the document]
Comment (sunnywu) [reply docs/guides/mobile-plugin-gma-android.md]: fixed

**Pull request merged**